### ISSUE TITLE
Add functions for months and years

### ISSUE THREAD

**Issue by DevL** [magplus/good_times#4]
Add similar convenience functions to deal with months and years.
## Examples
```elixir
3 |> months_ago # => {{2014, 12, 28}, {14, 25, 47}}
10 |> years_from_now # => {{2025, 02, 28}, {14, 25, 47}}
a_year_ago # => {{2014, 02, 28}, {14, 25, 47}}
a_month_from_now # => {{2015, 3, 28}, {14, 25, 47}}
```

**Comment by martinsvalin** [comment]
Years are most easily handled by actually adding/subtracting to the year part of the date tuple, instead of counting seconds. There are leap seconds and what not, so a simple `@seconds_per_year 365.25 * 24 * 60 * 60` might not cut it. (Imagine `a_year_after {{2015, 2, 2}, {0, 0, 0}}` producing `{2016, 2, 1}, {23, 59, 59}` because of some stupid leap second.)

Months are a bit trickier, and will have a bunch of logic to check if the day is greater than the last day of month etc.

**Comment by DevL** [comment]
We could handled months by changing the month portion of the date and then check if it's a valid date. If it isn't, decrease the day count by one. Rince and repeat until it's valid.

**Comment by martinsvalin** [comment]
No need to iterate. If day > ldom, use ldom.
`:calendar.last_day_of_the_month(2015, 2) #-> 28`
`:calendar.last_day_of_the_month(2016, 2) #-> 29`

**Comment by DevL** [comment]
True, that's much better.

**Comment by martinsvalin** [comment]
Closed by #15, #16 and #17.
